### PR TITLE
New module: pipx_info

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Adds the module `pipx_info`

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
plugins/modules/packaging/language/pipx_info.py

### CODE REVIEW

File: plugins/modules/packaging/language/pipx_info.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+---
+module: pipx_info
+short_description: Rretrieves information about applications installed with pipx
+version_added: 5.6.0
+description:
+    - Retrieve details about Python applications installed in isolated virtualenvs using pipx.
+options:
+    name:
+        description:
+            - Name of an application with C(pipx).
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pipx_info.py:22]: Is `installed` missing here?
```suggestion
            - Name of an application installed with C(pipx).
```

File: plugins/modules/packaging/language/pipx_info.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    - See also the C(pipx) documentation at U(https://pypa.github.io/pipx/).
+author:
+    - "Alexei Znamensky (@russoz)"
+'''
+
+EXAMPLES = '''
+'''
+
+RETURN = '''
+'''
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pipx_info.py:60]: You should also add some examples and describe the return values.

File: plugins/modules/packaging/language/pipx_info.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        ),
+        supports_check_mode=True,
+    )
+
+    def __init_module__(self):
+        if self.vars.executable:
+            self.command = [self.vars.executable]
+        else:
+            facts = ansible_facts(self.module, gather_subset=['python'])
+            self.command = [facts['python']['executable'], '-m', 'pipx']
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pipx_info.py:88]: Why not use [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable)? (Also exists in Python 2.)

File: plugins/modules/packaging/language/pipx_info.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    executable:
+        description:
+            - Path to the C(pipx) installed in the system.
+            - >
+              If not specified, the module will use C(python -m pipx) to run the tool,
+              using the same Python interpreter as ansible itself.
+        type: path
+notes:
+    - This module does not install the C(pipx) python package, however that can be easily done with the module M(ansible.builtin.pip).
+    - This module does not require C(pipx) to be in the shell C(PATH), but it must be loadable by Python as a module.
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pipx_info.py:49]: Out of curiosity: why is that required?

**Review by felixfontein — COMMENTED** [review]
Thanks for this contribution :)

File: plugins/modules/packaging/language/pipx_info.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/packaging/language/pipx_info.py:60]: yup, it's on the to-do list

File: plugins/modules/packaging/language/pipx_info.py (hunk 3 above)
Comment (russoz) [reply plugins/modules/packaging/language/pipx_info.py:88]: That part was just copied from the `pipx` module. As of Python 2 compatibility, `pipx` requires Python 3.6+, so that is not a concern.

File: plugins/modules/packaging/language/pipx_info.py (hunk 4 above)
Comment (russoz) [reply plugins/modules/packaging/language/pipx_info.py:49]: Because we invoke `pipx` as `<python> -m pipx`, allowing `pipx` installed both at system-wide level and user-wide level. In fact, in the integration test we install `pipx` with:
```yaml
- name: install pipx
  pip:
    name: pipx
    extra_args: --user
```

Comment (russoz) [reply plugins/modules/packaging/language/pipx_info.py:49]: Because we invoke `pipx` as `<python> -m pipx`, allowing `pipx` installed both at system-wide level and user-wide level. In fact, in the integration test we install `pipx` with:
```yaml
- name: install pipx
  pip:
    name: pipx
    extra_args: --user
```

File: plugins/modules/packaging/language/pipx_info.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/packaging/language/pipx_info.py:124]: done

**Comment by Andersson007** [comment]
@russoz could you please check your matrix? I also tried to use your email to ping you but haven't got a response:) thanks:) Excuse me folks

**Comment by russoz** [comment]
> @russoz could you please check your matrix? I also tried to use your email to ping you but haven't got a response:) thanks:) Excuse me folks

I wasn't fast enough ;)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks a lot for contributing this!
